internal/connect/eula.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return map[string]string{}, err
+	}
+	url, err := base.Parse(eulaIndex)
+	if err != nil {
+		return map[string]string{}, err
+	}
+	Debug.Printf("Downloading license index from %s...", url)
+	index, err := downloadFile(url.String())
+	if err != nil {
+		return map[string]string{}, err
+	}
+	ret, err := parseEULAIndex(index, baseURL)
+	if err != nil {
+		return map[string]string{}, err
+	}
+	Debug.Printf("Downloaded license index %+v", ret)
+	return ret, nil
+}
+
+func showTextInPager(text []byte) error {
+	// POSIX default
+	pager := "more"
+	if p, ok := os.LookupEnv("PAGER"); ok && p != "" {
+		pager = p
+	}
+	Debug.Printf("Using %s as pager.", pager)
+
+	// write temp file with text
+	tmpfile, err := ioutil.TempFile("", "connect")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpfile.Name())
+
+	if _, err := tmpfile.Write(text); err != nil {
+		return err
+	}
+	if err := tmpfile.Close(); err != nil {
+		return err
+	}
+
+	// run interactive pager
+	comm := exec.Command(pager, tmpfile.Name())
+	comm.Stdin = os.Stdin
+	comm.Stdout = os.Stdout
+	comm.Stderr = os.Stderr
+	return comm.Run()
+}
+
+// AcceptEULA displays EULA and prompts for acceptance (or does nothing if there's no EULA)
+func AcceptEULA(autoAgree bool) error {
+	// separate EULAs are only for addon products
+	if CFG.Product.isEmpty() {
+		return nil
+	}
+
+	// fetch list of extensions and search for requested product
+	base, err := baseProduct()
+	if err != nil {
+		return err
+	}
+	prod, err := showProduct(base)
+	if err != nil {
+		return err
+	}
+	extension, err := prod.findExtension(CFG.Product)
+	// no EULA or product not found (handle in registration code)
+	if strings.TrimSpace(extension.EULAURL) == "" {
+		return nil
+	}
+
+	eulas, err := downloadEULAIndex(extension.EULAURL)
+	if err != nil {
+		return err
+	}
+	lang := selectEULALanguage(eulas)
+	// this should happen only if there are no license files in index
+	if lang == "" {
+		return fmt.Errorf("No EULAs found at: %s", extension.EULAURL)
+	}
+
+	// download EULA text
+	eula, err := downloadFile(eulas[lang])
+	// trim BOM (some pagers display it as hex codes)
+	eula = bytes.TrimLeft(eula, "\xef\xbb\xbf")
+	// add header
+	header := fmt.Sprintf(
+		"In order to install '%s', you must agree to terms of the following license agreement:\n\n",
+		extension.FriendlyName)
+	eula = append([]byte(header), eula...)
+	if !autoAgree {
+		// TODO: maybe also add navigation and exit hints like in
+		// https://github.com/openSUSE/zypper/blob/master/src/utils/pager.cc
+		err = showTextInPager(eula)
+	}
+	// print EULA also on pager failure
+	if autoAgree || err != nil {
+		// there is some padding at the end which doesn't make much sense outside of pager
+		Info.Println(string(bytes.TrimSpace(eula)))
+	}
+	// license was presented, nothing more to do in auto mode
+	if autoAgree {
+		// TODO: maybe print some message e.g. "License accepted (auto confirmation)"
+		return nil
+	}
+	// prompt for acceptance
```

Comment:
why check `os.Stdout`?

Reply:
Because it maybe called from a script and there is no user there to type yes or no.
`./out/suseconnect  -p sle-ha/15.3/x86_64 > /tmp/log`  just stops.
Also this wrapper writes to a pipe
https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/6d9020ef021741bee4eaadec4a36a7d4246cab6c/usr/sbin/registercloudguest#L433-L445

Reply:
I'm not sure if we need to do so much automation around this.
Try `zypper rm python > /dev/null`. It doesn't print question, hangs, but still accepts answer from stdin.

Reply:
sure, but what if some customers are using SUSEConnect to register products in scripts that redirect stdout and have no user interaction.

Reply:
They will need to adapt. It will fail anyway if we detect the terminal.

Reply:
okay if we are going to deliberately break backward compatibility, then we should at least make it easier for customers to debug and fix by exiting with a useful message:
```
    if !isTerminal(os.stdout) {
        return fmt.Errorf("Standard output is not a console, please use --auto-agree-with-licenses")
    }

...

func isTerminal(f *os.File) bool {
    stat, _ := f.Stat()
    return (stat.Mode() & os.ModeCharDevice) == os.ModeCharDevice
}
```

Reply:
I like the general idea but I'm not sure if we will really cover all problem cases.
If not then maybe it's better to keep it simple and handle it in release notes or similar.
Most problematic clients will most probably be on our side anyway.